packages/mui-joy/src/Select/SelectProps.ts

```diff
@@ -104,7 +104,10 @@ export interface SelectOwnProps<TValue extends {}> extends SelectStaticProps {
   /**
    * Callback fired when an option is selected.
    */
-  onChange?: (value: TValue | null) => void;
+  onChange?: (
+    e: React.MouseEvent | React.KeyboardEvent | React.FocusEvent | null,
```

Comment:
💯  to remove these shorthands. As far as I know, these are the first in the codebase. I would argue that it's confusing for developers that are getting started, and less clear for experienced developers that are coming back to an old codebase. I think that e > event only for writing code, but if we assume that this code will be read more than written, then I think it would be better with:

```suggestion
    event: React.MouseEvent | React.KeyboardEvent | React.FocusEvent | null,
```

Reply:
All right, will do a follow-up PR.